scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -338,24 +344,43 @@ object Execution {
 
   private case class Mapped[S, T](prev: Execution[S], fn: S => T) extends Execution[T] {
     def runStats(conf: Config, mode: Mode, cache: EvalCache)(implicit cec: ConcurrentExecutionContext) =
-      cache.getOrElseInsert(this,
+      cache.getOrElseInsert(conf, this,
         prev.runStats(conf, mode, cache)
           .map { case (s, stats) => (fn(s), stats) })
   }
   private case class GetCounters[T](prev: Execution[T]) extends Execution[(T, ExecutionCounters)] {
     def runStats(conf: Config, mode: Mode, cache: EvalCache)(implicit cec: ConcurrentExecutionContext) =
-      cache.getOrElseInsert(this,
+      cache.getOrElseInsert(conf, this,
         prev.runStats(conf, mode, cache).map { case tc @ (t, c) => (tc, c) })
   }
   private case class ResetCounters[T](prev: Execution[T]) extends Execution[T] {
     def runStats(conf: Config, mode: Mode, cache: EvalCache)(implicit cec: ConcurrentExecutionContext) =
-      cache.getOrElseInsert(this,
+      cache.getOrElseInsert(conf, this,
         prev.runStats(conf, mode, cache).map { case (t, _) => (t, ExecutionCounters.empty) })
   }
 
+  private case class TransformedConfig[T](prev: Execution[T], fn: Config => Config) extends Execution[T] {
+    def runStats(conf: Config, mode: Mode, cache: EvalCache)(implicit cec: ConcurrentExecutionContext) = {
+      val mutatedConfig = fn(conf)
+      cache.getOrElseInsert(mutatedConfig, this, prev.runStats(mutatedConfig, mode, cache))
+    }
+  }
+
+  private case class WithNewCache[T](prev: Execution[T]) extends Execution[T] {
+    def runStats(conf: Config, mode: Mode, cache: EvalCache)(implicit cec: ConcurrentExecutionContext) = {
+      val ec = new EvalCache
+      val ret = prev.runStats(conf, mode, ec)
+      // When the final future in complete we stop the submit thread
```

Comment:
I forgot about this. This will create two concurrent cascading threads, which is not safe. We need to share that cascading thread between the two, I think.